File: gluecodium/src/test/resources/smoke/comments/output/android/com/example/smoke/ExcludedComments.java
/*

 *
 */

package com.example.smoke;

import android.support.annotation.NonNull;
import com.example.NativeBase;

/**
 * <p>This is some very useful class.
 * @hidden
 */
public final class ExcludedComments extends NativeBase {
    /**
     * <p>This is some very useful constant.
     * @hidden
     */
    public static final boolean VERY_USEFUL = true;
    /**
     * <p>This is some very useful enum.
     * @hidden
     */
    public enum SomeEnum {
        /**
         * <p>Not quite useful
         * @hidden
         */
        USELESS(0);
        /**
         * @hidden
         */
        public final int value;

        SomeEnum(final int value) {
            this.value = value;
        }
    }
    /**
     * <p>This is some very useful exception.
     * @hidden
     */
    public static final class SomethingWrongException extends Exception {
        /**
         * @hidden
         * @param error The error.
         */
        public SomethingWrongException(final ExcludedComments.SomeEnum error) {
            super(error.toString());
            this.error = error;
        }

        /**
         * @hidden
         */
        public final ExcludedComments.SomeEnum error;
    }
    /**
     * <p>This is some very useful struct.
     * @hidden
     */
    public static final class SomeStruct {
        /**
         * <p>How useful this struct is
         * remains to be seen
         * @hidden
         */
        public boolean someField;

        /**
         * <p>This is how easy it is to construct.
         * @param someField <p>How useful this struct is
         * remains to be seen
         */
        public SomeStruct(final boolean someField) {
            this.someField = someField;
        }


    }

    /**
     * @hidden
     */
    static class SomeLambdaImpl extends NativeBase implements SomeLambda {
        protected SomeLambdaImpl(final long nativeHandle, final Object dummy) {
            super(nativeHandle, new Disposer() {
                @Override
                public void disposeNative(long handle) {
                    disposeNativeHandle(handle);
                }
            });
        }

        private static native void disposeNativeHandle(long nativeHandle);


        /**
         * <p>This is some very useful lambda that does it.
         * @hidden
         * @param p0 <p>Very useful input parameter
         * @param index <p>Slightly less useful input parameter
         * @return <p>Usefulness of the input
         */



        public native double doIt(@NonNull final String p0, final int index);


    }

    /**
     * <p>This is some very useful lambda that does it.
     * @hidden
     */
    @FunctionalInterface
    public interface SomeLambda {
        /**
         * <p>This is some very useful lambda that does it.
         * @hidden
         * @param p0 <p>Very useful input parameter
         * @param index <p>Slightly less useful input parameter
         * @return <p>Usefulness of the input
         */



        double doIt(@NonNull final String p0, final int index);
    }


    /**
     * For internal use only.
     * @hidden
     * @param nativeHandle The SDK nativeHandle instance.
     * @param dummy The SDK dummy instance.
     */
    protected ExcludedComments(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }

    private static native void disposeNativeHandle(long nativeHandle);


    /**
     * <p>This is some very useful method that measures the usefulness of its input.
     * @hidden
     * @param inputParameter <p>Very useful input parameter
     * @return <p>Usefulness of the input
     * @throws ExcludedComments.SomethingWrongException <p>Sometimes it happens.
     */



    public native boolean someMethodWithAllComments(@NonNull final String inputParameter) throws ExcludedComments.SomethingWrongException;

    /**
     * <p>This is some very useful method that does nothing.
     * @hidden
     */



    public native void someMethodWithoutReturnTypeOrInputParameters();

    /**
     * <p>Gets some very useful property.
     * @hidden
     * @return <p>Some very useful property.
     */



    public native boolean isSomeProperty();

    /**
     * <p>Sets some very useful property.
     * @hidden
     * @param value <p>Some very useful property.
     */



    public native void setSomeProperty(final boolean value);



}

File: gluecodium/src/test/resources/smoke/comments/output/android-kotlin/com/example/smoke/ExcludedComments.kt
/*

 *
 */

@file:JvmName("ExcludedCommentsExtensions")


package com.example.smoke

import com.example.NativeBase

/**
 * This is some very useful class.
 * @suppress
 */
class ExcludedComments : NativeBase {

    /**
     * This is some very useful enum.
     * @suppress
     */
    enum class SomeEnum(@JvmField val value: Int) {
        /**
         * Not quite useful
         * @suppress
         */
        USELESS(0);
    }
    /**
     * This is some very useful exception.
     * @suppress
     */
    class SomethingWrongException(@JvmField val error: ExcludedComments.SomeEnum) : Exception(error.toString())


    /**
     * This is some very useful struct.
     * @suppress
     */
    class SomeStruct {
        /**
         * How useful this struct is
         * remains to be seen
         * @suppress
         */
        @JvmField var someField: Boolean



        /**
         * This is how easy it is to construct.
         * @param someField How useful this struct is
         * remains to be seen
         */
        constructor(someField: Boolean) {
            this.someField = someField
        }





    }

    /**
     * This is some very useful lambda that does it.
     * @suppress
     */
    fun interface SomeLambda {
        /**
         * This is some very useful lambda that does it.
         * @suppress
         * @param p0 Very useful input parameter
         * @param p1 Slightly less useful input parameter
         * @return Usefulness of the input
         */

        fun apply(p0: String, p1: Int) : Double
    }

    /**
     * @suppress
     *
     * This class is used to represent C++ implementations of the interface or lambda in Kotlin.
     * It is instantiated by JNI and should not be used by the end users.
     */
    private class SomeLambdaImpl : NativeBase, SomeLambda {
        protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
            : super(nativeHandle, { disposeNativeHandle(it) }) {}

        /**
         * This is some very useful lambda that does it.
         * @suppress
         * @param p0 Very useful input parameter
         * @param p1 Slightly less useful input parameter
         * @return Usefulness of the input
         */

        override external fun apply(p0: String, p1: Int) : Double



        companion object {
            @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
        }
    }


    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}



    /**
     * This is some very useful method that measures the usefulness of its input.
     * @suppress
     * @param inputParameter Very useful input parameter
     * @return Usefulness of the input
     * @throws ExcludedComments.SomethingWrongException Sometimes it happens.
     */
    @Throws(ExcludedComments.SomethingWrongException::class)
    external fun someMethodWithAllComments(inputParameter: String) : Boolean
    /**
     * This is some very useful method that does nothing.
     * @suppress
     */

    external fun someMethodWithoutReturnTypeOrInputParameters() : Unit

    /**
     * Some very useful property.
     * @suppress
     */
    var isSomeProperty: Boolean
        external get
        external set




    companion object {
        /**
         * This is some very useful constant.
         * @suppress
         */
        @JvmField final val VERY_USEFUL: Boolean = true
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}


